src/QPMotionConstr.cpp

```diff
@@ -113,13 +113,16 @@ MotionConstrCommon::ContactData::ContactData(const rbd::MultiBody & mb,
 
 MotionConstrCommon::MotionConstrCommon(const std::vector<rbd::MultiBody> & mbs, int robotIndex)
 : robotIndex_(robotIndex), alphaDBegin_(-1), nrDof_(mbs[robotIndex_].nrDof()), lambdaBegin_(-1), fd_(mbs[robotIndex_]),
-  fullJacLambda_(), jacTrans_(6, nrDof_), jacLambda_(), cont_(), curTorque_(nrDof_), A_(), AL_(nrDof_), AU_(nrDof_)
+  fullJacLambda_(), jacTrans_(6, nrDof_), jacLambda_(), cont_(), curTorque_(nrDof_), lastTorque_(nrDof_), A_(),
+  AL_(nrDof_), AU_(nrDof_)
 {
   assert(std::size_t(robotIndex_) < mbs.size() && robotIndex_ >= 0);
+  lastTorque_.setZero(); // TODO should be initialized with a gravity-compensation torque?
 }
 
 void MotionConstrCommon::computeTorque(const Eigen::VectorXd & alphaD, const Eigen::VectorXd & lambda)
 {
+  lastTorque_ = curTorque_; // NOTE: this fuction should not be called multiple times within one control cycle
```

Comment:
Indeed there is:
- no guarantee that `computeTorque` is called at all (but this can be handled at mc_rtc level)
- no guarantee is it not called multiple time

For that second point I suggest we introduce two counters to `MotionConstrCommon`:
- `updateIter_`: starts at 0 and is incremented every time `::update`) is called
- `lastTorqueIter_`: starts at 0, when `computeTorque` is called if `updateIter_ != lastTorqueIter_` then we update `lastTorque_` and `lastTorqueIter_` otherwise we don't do anything

Reply:
thanks for the feedback, this is solved with my latest commit

Reply:
I don't think any new commit since my own patch, maybe the push didn't get through because you didn't rebase the branch with these patches:
```
# replace ndehio with the name of your fork remote
git pull --rebase ndehio topic/TorqueDerivativeConstraint
git push
```